integration-tests/tests/src/tests/sets.ts

```diff
@@ -0,0 +1,493 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
+
+describe("Sets", () => {
+  beforeEach(Realm.clearTestState);
+
+  const IntSetObjectSchema = {
+    name: "IntSetObject",
+    properties: {
+      intSet: "int<>",
+    },
+  };
+  interface IntSetObject {
+    intSet: Realm.Set<Realm.Types.Int>;
+  }
+
+  const teamSchema = {
+    name: "Team",
+    properties: {
+      names: "string<>",
+    },
+  };
+  interface Team {
+    names: Realm.Set<string>;
+  }
+
+  describe("Schema", () => {
```

Comment:
This could later be removed, as the tests for parsing schemas should cover this.

Reply:
We would need some sort of reminder.  We should be tracking what should later be refactored.

Reply:
Would a TODO comment be sufficient or is it better to have something specific to this test refactoring, TOREFACTOR or something? @kraenhansen

Reply:
`// TODO` is fine 🙂
(alternatively TOREDO? 🙈 )

Reply:
I guess I can just TODO then, it'll all be in same directory so should be pretty easy to figure out.